manifests/profile/neutron/server.pp

```diff
@@ -12,11 +12,40 @@
   $tunnel_id_ranges              = $::openstack::config::neutron_tunnel_id_ranges # ['1:1000']
   $controller_management_address = $::openstack::config::controller_address_management
 
-  class  { '::neutron::plugins::ml2':
-    type_drivers         => $type_drivers,
-    tenant_network_types => $tenant_network_type,
-    mechanism_drivers    => $mechanism_drivers,
-    tunnel_id_ranges     => $tunnel_id_ranges
+  if ($::openstack::config::neutron_core_plugin == 'ml2') {
```

Comment:
The ml2 core plugin is called 'neutron.plugins.ml2.plugin.Ml2Plugin', not simply 'ml2' https://github.com/puppetlabs/puppetlabs-openstack/blob/master/manifests/config.pp#L57